frappe/public/js/frappe/list/list_sidebar.js

```diff
@@ -49,12 +48,27 @@ frappe.views.ListSidebar = class ListSidebar {
 
 	setup_views() {
 		var show_list_link = false;
+		var cur_doc=this.doctype;
+		var check = false;
+		frappe.call({
+			method: "frappe.core.page.calendar.calendar.get_all_calendars",
+			type: "GET"
+		}).then(r => {
+			$.each(r["message"], function( index, value ) {
+				if(cur_doc == value){
+					check = true;
+					return false;
+				}
+			});
+			if (check) {
+				this.sidebar.find('.list-link[data-view="Calendar"]').removeClass("hide");
+				this.sidebar.find('.list-link[data-view="Gantt"]').removeClass('hide');
+				show_list_link = true;
+				this.setup_calendar_view();
```

Comment:
there's no need for the variable check, just put this inside the if condition?

Reply:
yeah removed